source/docs/software/commandbased/commands.rst

```diff
@@ -1,244 +1,124 @@
-or it was interrupted (either by another command or by being explicitly
-canceled). The method argument specifies the manner in which the command
-ended; users can use this to differentiate the behavior of their command
-end accordingly. The end block should be used to “wrap up” command state
-in a neat way, such as setting motors back to zero or reverting a
-solenoid actuator to a “default” state.
+Simple Command Example
+----------------------
 
-Specifying end conditions
-~~~~~~~~~~~~~~~~~~~~~~~~~
+What might a functional command look like in practice? As before, below is a simple command from the HatchBot example project (`Java <https://github.com/wpilibsuite/allwpilib/tree/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/hatchbottraditional>`__, `C++ <https://github.com/wpilibsuite/allwpilib/tree/master/wpilibcExamples/src/main/cpp/examples/HatchbotTraditional>`__) that uses the ``HatchSubsystem`` introduced in the previous section:
 
-.. code-block:: java
+.. tabs::
 
-   @Override
-   public boolean isFinished() {
-     // This return value will specify whether the command has finished!  The default is "false," which will make the
-     // command never end.
-     return false;
-   }
+  .. tab:: Java
 
-Just like ``execute()``, the ``isFinished()`` method of the command is
-called repeatedly, whenever the scheduler’s ``run()`` method is called.
-As soon as it returns true, the command’s ``end()`` method is called and
-it is un-scheduled. The ``isFinished()`` method is called *after* the
-``execute()`` method, so the command *will* execute once on the same
-iteration that it is un-scheduled.
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/hatchbottraditional/commands/GrabHatch.java
+      :language: java
+      :lines: 8-
+      :linenos:
+      :lineno-start: 8
 
-Simple command example
-----------------------
+  .. tab:: C++ (Header)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/examples/HatchbotTraditional/include/commands/GrabHatch.h
+      :language: c++
+      :lines: 8-
+      :linenos:
+      :lineno-start: 8
+
+  .. tab:: C++ (Source)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/examples/HatchbotTraditional/cpp/commands/GrabHatch.cpp
+      :language: c++
+      :lines: 8-
+      :linenos:
+      :lineno-start: 8
+
+Notice that the hatch subsystem used by the command is passed into the command through the command’s constructor. This is a pattern called `dependency injection <https://en.wikipedia.org/wiki/Dependency_injection>`__, and allows users to avoid declaring their subsystems as global variables. This is widely accepted as a best-practice - the reasoning behind this is discussed in a :doc:`later section <structuring-command-based-project>`.
```

Comment:
I think it is confusing to call this dependency injection because when I hear DI, I think Guice.  I think a better word for this pattern is `Aggregation`.

Reply:
I've always heard of it as Dependency Injection.

Reply:
I've never heard people call this "aggregation" either. Furthermore, high school students have never heard of dependency injection frameworks, so they wouldn't be able to confuse the phrase "dependency injection" with anything.

Reply:
DI is the correct term; DI container frameworks are one way of implementing DI, but the concept is significantly more-general than that.  Additionally, if teams *really want* to implement this DI with a container framework, they can ;)